src/back/glsl/mod.rs

```diff
@@ -1771,13 +1771,20 @@ impl<'a, W: Write> Writer<'a, W> {
             // keyword which ceases all further processing in a fragment shader, it's called OpKill
             // in spir-v that's why it's called `Statement::Kill`
             Statement::Kill => writeln!(self.out, "{}discard;", level)?,
-            // Issue an execution or a memory barrier.
+            // Issue a memory barrier. Please note that to ensure visibility,
+            // OpenGL always requires a call to the `barrier()` function after a `memoryBarrier*()`
             Statement::Barrier(flags) => {
-                if flags.is_empty() {
-                    writeln!(self.out, "{}barrier();", level)?;
-                } else {
-                    writeln!(self.out, "{}groupMemoryBarrier();", level)?;
+                if flags.contains(crate::Barrier::STORAGE) {
+                    writeln!(self.out, "{}memoryBarrierBuffer();", level)?;
+                    writeln!(self.out, "{}memoryBarrierImage();", level)?;
+                    writeln!(self.out, "{}memoryBarrierAtomicCounter();", level)?;
```

Comment:
I don't think this is necessary, it seems to be affecting the stuff we aren't exposing in WebGPU - https://www.lighthouse3d.com/tutorials/opengl-atomic-counters/